### PR TITLE
Added ability to edit patient details

### PR DESCRIPTION
**Pull request by jmcameron**
The patient edit form allows the user to select a patient (using the 'find-patient' directive) and then switches to editng the selected patient.

All editing functionality is impemented except for swapping the debitor group.

This update includes an update to the 'location-select' directive that supports an 'initial-location' attribute.  Using the existing 'defaultVillage' attribute did not seem to work right for initial location values so I added the new option for 'initial-location' attribute.  This should be backwards compatible, although it may be possible to merge the functionality and only have one.

### CODE REVIEW

File: client/src/partials/patient_edit/edit_patient.html [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+      <div class="col-xs-5">
+      </div> <!-- End column 5 row -->
+    </div> <!-- end row -->
+  </div>
+</main>
+
+<footer>
+  <div class="pull-right">
+    <a id="submitPatient" ng-click="updatePatient()"
+       ng-disabled="session.failedSessionValidation || location.$invalid || details.$invalid"
```
Comment (jniles) [inline client/src/partials/patient_edit/edit_patient.html]: Is it possible to take the `$invalid` logic like this out of the template?  Maybe include them as part of the session.failedSessionValidation check?

File: client/src/partials/patient_edit/edit_patient.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  '$routeParams',  
+  '$translate',
+  'connect',
+  'validate',
+  'messenger',
+  'util',
+  'appstate',
+  function ($scope, $routeParams, $translate, connect, validate, messenger, util, appstate) {
+    var dependencies = {},
+        original_patient_data = null,
```
Comment (jniles) [inline client/src/partials/patient_edit/edit_patient.js]: For JS variables, we prefer to us mixedCase.  Unfortunately, the database is constructed using underlines_not_spaces syntax, so we cannot get away from the underlines in some cases.  Could you change `original_patient_data` and `editable_fields`, etc, to mixedCase?

File: client/src/partials/patient_edit/edit_patient.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    $scope.initialiseEditing = function initialiseEditing (selectedPatient) {
+      if (selectedPatient && 'uuid' in selectedPatient && selectedPatient.uuid) {
+        patient_uuid = $scope.patient_uuid = selectedPatient.uuid;
+	dependencies.patient.query.where[0] = 'patient.uuid=' + patient_uuid;
+	validate.process(dependencies)
+	  .then(startup);
+	session.mode = 'edit';
+      }
+      else {
+	messenger.danger($translate('ERR_MISSING'));
```
Comment (jniles) [inline client/src/partials/patient_edit/edit_patient.js]: I cannot find a translation for this?  Could you add a namespaced translation in en.json?

File: client/src/partials/patient_edit/edit_patient.html (hunk 1 above)
Comment (jmcameron) [inline client/src/partials/patient_edit/edit_patient.html]: I do not know how to access the 'location' form from within the 'patientEdit' controller.  I'll fix the other things and then get with you to do this later as a separate pull request.  I have a few other cleanup things that I can do at the same time.

File: client/src/partials/patient_edit/edit_patient.js (hunk 3 above)
Comment (jmcameron) [inline client/src/partials/patient_edit/edit_patient.js]: I did not realize that 'ERR_MISSING' was not translated.  I copied it from 'swap_debtor'.
I will add a translation for this, but I'm having second thoughts on exposing this internal and hopefully-rare error to end users.   If we decide not to expose this, I'll add it to my next round of improvements on the form (in another pull request).

File: client/src/partials/patient_edit/edit_patient.html (hunk 1 above)
Comment (jniles) [inline client/src/partials/patient_edit/edit_patient.html]: Okay, sounds good.

File: client/src/partials/patient_edit/edit_patient.js (hunk 3 above)
Comment (jniles) [inline client/src/partials/patient_edit/edit_patient.js]: Sounds like a discussion we should all have as a team.  We can add that discussion to the agenda.

**Pull request merged**